tests/system-tests/internal/supporttools/tcpdump.go

```diff
@@ -0,0 +1,543 @@
+
+	var err error
+
+	glog.V(100).Infof("Checking support-tools deployment don't exist")
+
+	err = apiobjectshelper.DeleteDeployment(apiClient, stDeploymentName, stNamespace)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to delete deployment %s from namespace %s",
+			stDeploymentName, stNamespace)
+	}
+
+	glog.V(100).Infof("Sleeping 10 seconds")
+	time.Sleep(10 * time.Second)
+
+	glog.V(100).Infof("Removing ServiceAccount")
+
+	err = apiobjectshelper.DeleteServiceAccount(apiClient, supportToolsDeploySAName, stNamespace)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to remove serviceAccount %q from namespace %q",
+			supportToolsDeploySAName, stNamespace)
+	}
+
+	glog.V(100).Infof("Creating ServiceAccount")
+
+	err = apiobjectshelper.CreateServiceAccount(apiClient, supportToolsDeploySAName, stNamespace)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create serviceAccount %q in namespace %q",
+			supportToolsDeploySAName, stNamespace)
+	}
+
+	glog.V(100).Infof("Removing Cluster RBAC")
+
+	err = apiobjectshelper.DeleteClusterRBAC(apiClient, supportToolsDeployRBACName)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to delete supporttools RBAC %q", supportToolsDeployRBACName)
+	}
+
+	glog.V(100).Infof("Creating Cluster RBAC")
+
+	err = apiobjectshelper.CreateClusterRBAC(apiClient, supportToolsDeployRBACName, supportToolsRBACRole,
+		supportToolsDeploySAName, stNamespace)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create supporttools RBAC %q in namespace %s",
+			supportToolsDeployRBACName, stNamespace)
+	}
+
+	glog.V(100).Infof("Defining container configuration")
+
+	deployContainer := defineTCPDumpContainer(
+		stImage,
+		packetCaptureInterface,
+		packetCaptureProtocol)
+
+	glog.V(100).Infof("Obtaining container definition")
+
+	deployContainerCfg, err := deployContainer.GetContainerCfg()
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain container definition: %w", err)
+	}
+
+	glog.V(100).Infof("Defining deployment configuration")
+
+	deployLabelsMap := map[string]string{
+		strings.Split(stDeploymentLabel, "=")[0]: strings.Split(stDeploymentLabel, "=")[1]}
+
+	stDeployment := defineDeployment(
+		apiClient,
+		deployContainerCfg,
+		stDeploymentName,
+		stNamespace,
+		supportToolsDeploySAName,
+		scheduleOnHosts,
+		deployLabelsMap)
+
+	glog.V(100).Infof("Creating deployment")
+
+	stDeployment, err = stDeployment.CreateAndWaitUntilReady(5 * time.Minute)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create deployment %s in namespace %s: %w",
+			stDeploymentName, stNamespace, err)
+	}
+
+	if stDeployment == nil {
+		return nil, fmt.Errorf("failed to create deployment %s", stDeploymentName)
+	}
+
+	return stDeployment, err
+}
+
+func defineTCPDumpContainer(
+	cImage,
+	packetCaptureInterface,
+	packetCaptureProtocol string) *pod.ContainerBuilder {
+	cName := "tcpdump"
+
+	cmd := tcpCaptureScript
+	if packetCaptureProtocol == udpProtocol {
+		cmd = udpCaptureScript
+	}
```

Comment:
imho this part should be done outside of _defineTCPDumpContainer_ and method should accept command as a parameter rather then reference global constants

Reply:
good idea! we will be more flexible in this case